src/dag.rs

```diff
@@ -165,10 +165,14 @@ pub struct IpldDag<Types: RepoTypes> {
 }
 
 impl<Types: RepoTypes> IpldDag<Types> {
+    /// Creates a new `IpldDag` for DAG operations.
     pub fn new(ipfs: Ipfs<Types>) -> Self {
         IpldDag { ipfs }
     }
```

Comment:
This feels a bit like an oversight for anyone having to write or read an "MSDN style" piece of documentation.

Looking that https://github.com/rs-ipfs/rust-ipfs/blob/3fc24b0705a2340bc04647729de875d3abd9edce/src/lib.rs#L387-L389 exists I am thinking this might be accidentially `pub fn`. While the current API is of course far from perfect, perhaps I'd like to suggest this:

```suggestion
    /// Creates a new `IpldDag` for DAG operations.
    // FIXME: duplicates Ipfs::dag(), having both is unnecessary
    pub fn new(ipfs: Ipfs<Types>) -> Self {
        IpldDag { ipfs }
    }
```